### PR TITLE
1.7.0 with Namada

### LINKED ISSUE #11: Move back to mainstream tendermint-rs
After moving Namada tendermint-rs back to the mainstream, we can remove the redundant tendermint handling for our forked `tendermint-rs`

### PR DESCRIPTION
**Pull request by yito88**
closes #11 

- Add `Namada` to `ChainConfig`
  - Reuse `CosmosSdkConfig` for generating Tendermint light clients
- Add `NamadaChain` as `ChainEndpoint` implementation
  - See `crates/relayer/src/chain/namada.rs` and files under `crates/relayer/src/chain/namada`
  - To submit Namada transactions and query
- Store Namada key to `KeyRing` #15

### CODE REVIEW

File: .github/workflows/cargo-doc.yaml [hunk 1]
```diff
@@ -20,7 +20,11 @@ jobs:
         with:
           toolchain: nightly-2023-07-13
           override: true
-
+      - name: Install Protoc
```
Comment (cwgoes) [inline .github/workflows/cargo-doc.yaml:23]: do we really need to ask informal to make this a dependency?

File: crates/relayer/src/chain/namada.rs [hunk 2]
```diff
@@ -0,0 +1,1235 @@
...
+        let proto_msgs = tracked_msgs.messages();
+        if proto_msgs.is_empty() {
+            return Ok(vec![]);
+        }
+        let mut tx_sync_results = vec![];
+        for msg in proto_msgs.iter() {
+            let response = self.send_tx(msg)?;
+
+            // Note: we don't have any height information in this case. This hack will fix itself
+            // once we remove the `ChainError` event (which is not actually an event)
```
Comment (cwgoes) [inline crates/relayer/src/chain/namada.rs:328]: is this our hack or informal's hack?

**Review by cwgoes — COMMENTED** [review]
Just took a quick look for now. Do you know if any other non-Cosmos-SDK chains have been added to upstream Hermes yet? Do they have any guidelines for new chains who want to submit PRs?

File: .github/workflows/cargo-doc.yaml (hunk 1 above)
Comment (yito88) [reply .github/workflows/cargo-doc.yaml:23]: We can't build Hermes importing `namada_sdk` and `namada` without `protoc`.

File: crates/relayer/src/chain/namada.rs (hunk 2 above)
Comment (yito88) [reply crates/relayer/src/chain/namada.rs:328]: This is informal's hack.
https://github.com/heliaxdev/hermes/blob/21e7fd266f3e2440b726fc02882e146144ec333d/crates/relayer/src/chain/cosmos/batch.rs#L218C27-L219

**Comment by yito88** [comment]
> Do you know if any other non-Cosmos-SDK chains have been added to upstream Hermes yet?
There are only Cosmos-SDK chains using `CosmosSdkChain` in the upstream. `NamadaChain` is the first one of non-Cosmos-SDK.

> Do they have any guidelines for new chains who want to submit PRs?
They said we could open a PR for Namada. I will share this progress and ask them about it again.

**Pull request closed without merging**